src/react-modules/FileUpload.tsx

```diff
@@ -0,0 +1,102 @@
+import { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTrash } from "@fortawesome/free-solid-svg-icons";
+import "./../FileUpload.css";
+
+export const FileUploadFull = ({
+  text,
+  personaIndex,
+  element,
+  handleJSONUpdate,
+}: {
+  text: string;
+  personaIndex: number;
+  element: string[];
+  handleJSONUpdate: (
+    personaIndex: number,
+    element: string[],
+    value: string | null
+  ) => void;
+}) => {
+  const [preview, setPreview] = useState<string | null>(null);
+
+  // To-Do: Impliment a feature to show the preview directly from the JSON data
+  // useEffect(() => {
+  //   setPreview(showcaseJSON.personas[personaIndex])
+  // },[element])
+
+  const convertBase64 = (file: File) => {
+    return new Promise((resolve, reject) => {
+      const fileReader = new FileReader();
+      fileReader.readAsDataURL(file);
+      fileReader.onload = () => {
+        resolve(fileReader.result);
+      };
+      fileReader.onerror = (error) => {
+        reject(error);
+      };
+    });
+  };
+
+  const handleChange = async (newValue: File | null) => {
+    let objectUrl = null;
```

Comment:
This from the exiting code; but why have a var? It is only used in the first branch of the expression. So could just be a local const there

Reply:
fixed on nextjs PR

```
 const handleChange = async (newValue: File | null) => {
    if (newValue) {
      try {
        const base64 = await convertBase64(newValue);
        if (typeof base64 === 'string') {
          setPreview(base64);
          handleJSONUpdate(element, base64);
        }
      } catch (error) {
        console.error('Error converting file:', error);
      }
    } else {
      setPreview(null);
      handleJSONUpdate(element, "");
    }
  };
```